### PR TITLE
[Core] Adding filtered accum reduction

### PR DESCRIPTION
**Pull request by ddiezrod**
**📝 Description**
This PR adds a filtered accum reduction derived from standard AccumReduction that lets us filter which elements we want to add to our vector returning a pair <bool, TDataType>.

Test has been added.

FYI @jrubiogonzalez

### CODE REVIEW

**Comment by jrubiogonzalez** [comment]
This looks great to me

**Comment by loumalouomega** [comment]
> This looks great to me

1hhv9m

**Review by loumalouomega — DISMISSED** [review]

**Review by loumalouomega — APPROVED** [review]

File: kratos/tests/cpp_tests/utilities/test_parallel_utilities.cpp [hunk 1]
```diff
@@ -332,6 +332,31 @@ KRATOS_TEST_CASE_IN_SUITE(AccumReductionSet, KratosCoreFastSuite)
...
+    std::iota(expct_data_vector.begin(), expct_data_vector.end(), 1);
+    std::transform(
+        expct_data_vector.begin(),
+        expct_data_vector.end(),
+        expct_data_vector.begin(),
+        [] (int x) { return x * 2; });
+
+    auto assembled_vector = block_for_each<FilteredAccumReduction<int>>(input_data_vector, [](int& rValue) {
+        const bool add_value = rValue % 2 != 0;
+        return std::pair<bool, std::size_t>(add_value, rValue + 1);
```
Comment (philbucher) [inline kratos/tests/cpp_tests/utilities/test_parallel_utilities.cpp:351]:
```suggestion
    auto assembled_vector = block_for_each<AccumReduction<int>>(input_data_vector, [](int& rValue) {
        return rValue % 2 ? rValue + 1 : 0;
```

**Review by philbucher — CHANGES_REQUESTED** [review]
Sorry for blocking, but I am not sure this is the way to go

Check my suggestion, it is easier to use the original implementation

Can you show a more complex usecase where your implementation can shine?

Also asking for feedback from @RiccardoRossi

**Review by philbucher — APPROVED** [review]
I am stupid, I confused Sum with Accum :facepalm: 

No comments then, sorry for the noise, makes full sense

**Comment by matekelemen** [comment]
Guys, in a total of 1 hour this PR was opened, """reviewed""" and merged. I'm all for speeding PR reviews up but you must give others a chance to review PRs **especially if they're targeting Core**.

**Comment by ddiezrod** [comment]
> Guys, in a total of 1 hour this PR was opened, """reviewed""" and merged. I'm all for speeding PR reviews up but you must give others a chance to review **especially if you're targeting Core**.

You are completely right, I checked the "automatic merge" option without thinking about it, and then just later I realized I should have waited so everyone can see it. If you have any comment on this, please go ahead and I will change whatever is needed.

In any case, this was reviewed by three experienced people in the group, so I do not think we can say this was not """""reviewed""""".

**Comment by matekelemen** [comment]
> In any case, this was reviewed by three experienced people in the group, so I do not think we can say this was not """""reviewed""""".

I am by no means disputing the competency of the people who managed to take a look. However, core stuff should have a wider audience, otherwise we won't know what gets implemented and there's less chance of getting features properly adopted by others. 

I'm just upset that this PR (and god knows how many others) would have flown completely under my radar if the implementation committee wasn't tagged in it.

**Pull request merged**